comp/netflow/goflowlib/convert.go

```diff
@@ -74,6 +75,11 @@ func applyAdditionalFields(flow *common.Flow, additionalFields common.Additional
 		if applied {
 			// We replaced a field of common.Flow with an additional field, no need to keep it in the map
 			delete(additionalFields, destination)
+		} else {
+			if field, ok := fieldValue.([]byte); ok {
+				// Write []byte as hex string for readability
+				additionalFields[destination] = bytesToHexString(field)
```

Comment:
Is it safe to modify additionalFields while we are looping thought it? 
In general, it's better to avoid modifying the same list/map we are looping by create a new list/map.

Reply:
Discussions I saw (e.g [here](https://github.com/golang/go/issues/9926)) tend to agree on this being "legal" and safe, but might be safer to copy indeed
WDYT of moving this logic to the `buildpayload` step, so that additionalFields are copied and formatted at this step?

Reply:
Since we want bug fixes to be minimal during agent release.
I think it might be better to keep thing here for now.

> but might be safer to copy indeed

I would tend to play safe here and use a copy to store a new list of additionalFields.

Also, I noticed that we are deleting entries from the additionalFields map `delete(additionalFields, destination)` passed as function parameter, I think we should avoid that kind of side effect too (side effect of modifying something passed as parameter).

Reply:
Okay, on it thanks!